src/main/java/com/google/cloud/teleport/bigtable/BigtableToParquet.java

```diff
@@ -0,0 +1,227 @@
+    Pipeline pipeline = Pipeline.create(options);
+    BigtableIO.Read read =
+        BigtableIO.read()
+            .withProjectId(options.getBigtableProjectId())
+            .withInstanceId(options.getBigtableInstanceId())
+            .withTableId(options.getBigtableTableId());
+
+    // Do not validate input fields if it is running as a template.
+    if (options.as(DataflowPipelineOptions.class).getTemplateLocation() != null) {
+      read = read.withoutValidation();
+    }
+
+    /**
+     * Steps:
+     * 1) Read records from Bigtable.
+     * 2) Convert a Bigtable Row to a GenericRecord.
+     * 3) Write GenericRecord(s) to GCS in parquet format.
+     */
+    pipeline
+        .apply("Read from Bigtable", read)
+        .apply("Transform to Parquet", MapElements.via(new BigtableToParquetFn()))
+        .setCoder(AvroCoder.of(GenericRecord.class, BigtableRow.getClassSchema()))
+        .apply(
+            "Write to Parquet in GCS",
+            FileIO.<GenericRecord>write()
+                .via(ParquetIO.sink(BigtableRow.getClassSchema()))
+                .to(options.getOutputDirectory())
+                .withPrefix(options.getFilenamePrefix())
+                .withSuffix(".parquet")
+                .withNumShards(options.getNumShards()));
+
+    return pipeline.run();
+  }
+
+  /**
+   * Translates a {@link PCollection} of Bigtable {@link Row} to a {@link PCollection} of {@link
+   * GenericRecord}.
+   */
+  static class BigtableToParquetFn extends SimpleFunction<Row, GenericRecord> {
+    @Override
+    public GenericRecord apply(Row row) {
+      ByteBuffer key = ByteBuffer.wrap(toByteArray(row.getKey()));
+      List<BigtableCell> cells = new ArrayList<>();
+      for (Family family : row.getFamiliesList()) {
+        String familyName = family.getName();
+        for (Column column : family.getColumnsList()) {
+          ByteBuffer qualifier = ByteBuffer.wrap(toByteArray(column.getQualifier()));
+          for (Cell cell : column.getCellsList()) {
+            long timestamp = cell.getTimestampMicros();
+            ByteBuffer value = ByteBuffer.wrap(toByteArray(cell.getValue()));
+            cells.add(new BigtableCell(familyName, qualifier, timestamp, value));
+          }
+        }
+      }
+      return new GenericRecordBuilder(BigtableRow.getClassSchema())
+          .set("key", key)
+          .set("cells", cells)
+          .build();
+    }
+  }
+
+  /**
+   * Extracts the byte array from the given {@link ByteString} without copy.
+   *
+   * @param byteString A {@link ByteString} from which to extract the array.
+   * @return an array of byte.
+   */
+  private static byte[] toByteArray(final ByteString byteString) {
+    try {
+      ZeroCopyByteOutput byteOutput = new ZeroCopyByteOutput();
+      UnsafeByteOperations.unsafeWriteTo(byteString, byteOutput);
+      return byteOutput.bytes;
+    } catch (IOException e) {
+      return byteString.toByteArray();
+    }
+  }
```

Comment:
Why dont you simply make this method in BigtableToAvro.java as protected so you can reuse without duplicating?

Reply:
Done